gdshelpers/parts/waveguide.py

```diff
@@ -114,49 +105,23 @@ def add_arc(self, final_angle, radius, final_width=None, n_points=128, shortest=
         n_points = n_points if n_points else 128
 
-        self._current_port.set_port_properties(**kargs)
+        self._current_port.set_port_properties(**kwargs)
+        sample_points = max(int(abs(angle) / (np.pi / 2) * n_points), 2)
         final_width = final_width or self.width
 
         angle = normalize_phase(angle, zero_to_two_pi=True) - (0 if angle > 0 else 2 * np.pi)
 
-        if not np.isclose(radius, 0) and not np.isclose(angle, 0) and radius > 0:
-            if angle > 0:
-                circle_center = (-np.sin(self.angle) * radius, np.cos(self.angle) * radius) + self.current_port.origin
-                start_angle = -np.pi / 2 + self.angle
-            else:
-                circle_center = (np.sin(self.angle) * radius, -np.cos(self.angle) * radius) + self.current_port.origin
-                start_angle = np.pi / 2 + self.angle
-
-            end_angle = start_angle + angle
-
-            # Calculate the points needed for this angle
-            points = max(int(abs(end_angle - start_angle) / (np.pi / 2) * n_points), 2)
-
-            phi = np.linspace(start_angle, end_angle, points)
-            upper_radius_points = np.linspace(radius - self.width / 2, radius - final_width / 2, points)
-            upper_line_points = np.array([upper_radius_points * np.cos(phi),
-                                          upper_radius_points * np.sin(phi)]).T + circle_center
-
-            lower_radius_points = np.linspace(radius + self.width / 2, radius + final_width / 2, points)
-            lower_line_points = np.array([lower_radius_points * np.cos(phi),
-                                          lower_radius_points * np.sin(phi)]).T + circle_center
-
-            polygon = shapely.geometry.Polygon(np.concatenate([upper_line_points, lower_line_points[::-1, :]]))
-            self._segments.append((self._current_port.copy(), polygon, abs(angle) * radius))
-
-            endpoint = shapely.geometry.Point(radius * np.cos(end_angle) + circle_center[0],
-                                              radius * np.sin(end_angle) + circle_center[1])
-            self._current_port.origin = endpoint.coords[0]
-
-        self._current_port.width = final_width
-        self._current_port.angle += angle
-
-        # assert self._segments[-1][1].is_valid, \
-        #     'Invalid polygon generated: %s' % shapely.validation.explain_validity(self._segments[-1][1])
+        self.add_parameterized_path(
+            path=lambda t: [radius * np.sin(abs(angle) * t), np.sign(angle) * -radius * (np.cos(angle * t) - 1)],
+            path_function_supports_numpy=True,
+            path_derivative=lambda t: [radius * np.cos(abs(angle) * t) * abs(angle),
+                                       np.sign(angle) * radius * (np.sin(angle * t) * angle)],
+            width=lambda t: np.array(self.width) * (1 - t) + np.array(final_width) * t,
+            sample_points=sample_points, sample_distance=0)
 
         return self
 
```

Comment:
Probably you should also say something about the form of the `width` parameter in the docstring of add_parameterized_path...

Reply:
Thanks! I've added the description to the docstring